detection_rules/rule.py

```diff
@@ -677,8 +686,30 @@ def _post_dict_transform(self, obj: dict) -> dict:
 
     def add_related_integrations(self, obj: dict) -> None:
         """Add restricted field related_integrations to the obj."""
-        # field_name = "related_integrations"
-        ...
+        field_name = "related_integrations"
+        package_integrations = obj.get(field_name, [])
+
+        if self.metadata.integration:
+            packages_manifest = load_integrations_manifests()
+            current_stack_version = load_current_package_version()
+
+            if self.check_restricted_field_version(field_name):
+                if isinstance(self.data, QueryRuleData) and self.data.language != 'lucene':
+                    package_integrations = self.get_packaged_integrations(self.data.ast, packages_manifest)
+
+                    for package in package_integrations:
+                        package["version"] = find_least_compatible_version(
+                            package=package["package"],
+                            integration=package["integration"],
+                            current_stack_version=current_stack_version,
+                            packages_manifest=packages_manifest)
+
+                        # if integration is not a policy template remove
+                        policy_templates = packages_manifest[package["package"]][package["version"]]["policy_templates"]
+                        if package["integration"] not in policy_templates:
+                            del package["integration"]
```

Comment:
why delete this? Not sure if I understand this

Reply:
We caught this with Okta specifically. For some Okta rules, we have `event.dataset:okta.sytem` as an example. When this is parsed, Okta is the package and system would be the integration. When this is built out and reaches this part of the code here is the related_integrations value:

```
{'package':'okta','integration':'system','version':'1.3.0'}
```
Therefore, integration is removed so we now have: ```{'package':'okta','version':'1.3.0'}```